server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java

```diff
@@ -79,8 +110,19 @@
      * @return {@link Role}
      */
     @Get("/{id}")
-    public Role readRole(UUID id) {
-        return roleServices.read(id);
+    public Single<HttpResponse<Role>> readRole(UUID id) {
+        return Single.fromCallable(() -> {
+            Role result = roleServices.read(id);
+            if (result == null) {
+                throw new RoleNotFoundException("No role item for UUID");
+            }
+            return result;
+        })
+        .observeOn(Schedulers.from(eventLoopGroup))
+        .map(createdRole -> {
```

Comment:
Should we rename `createdRole` to improve readability?

Reply:
updated to userRole for better readability